kms.tf

```diff
@@ -0,0 +1,81 @@
+data "aws_partition" "current" {}
```

Comment:
should this be a submodule? can move it later

Reply:
It seems to cause more deferred resolution of the KMS key which can lead to full recreation of some resources (like EFS) during an update to an existing cluster. I don't know if that's a great reason...

Reply:
must be yet another plan vs apply thing.